scheduler/task.go

```diff
@@ -301,24 +301,31 @@ func (t *task) spin() {
 				} else {
 					consecutiveFailures = 0
 				}
-				if consecutiveFailures >= t.stopOnFailure {
-					taskLogger.WithFields(log.Fields{
-						"_block":               "spin",
-						"task-id":              t.id,
-						"task-name":            t.name,
-						"consecutive failures": consecutiveFailures,
-						"error":                t.lastFailureMessage,
-					}).Error(ErrTaskDisabledOnFailures)
-					// You must lock on state change for tasks
-					t.Lock()
-					t.state = core.TaskDisabled
-					t.Unlock()
-					// Send task disabled event
-					event := new(scheduler_event.TaskDisabledEvent)
-					event.TaskID = t.id
-					event.Why = fmt.Sprintf("Task disabled with error: %s", t.lastFailureMessage)
-					defer t.eventEmitter.Emit(event)
-					return
+				if t.stopOnFailure >= 0 {
+					if consecutiveFailures >= t.stopOnFailure {
+						taskLogger.WithFields(log.Fields{
+							"_block":               "spin",
+							"task-id":              t.id,
+							"task-name":            t.name,
+							"consecutive failures": consecutiveFailures,
+							"error":                t.lastFailureMessage,
+						}).Error(ErrTaskDisabledOnFailures)
+						// You must lock on state change for tasks
+						t.Lock()
+						t.state = core.TaskDisabled
+						t.Unlock()
+						// Send task disabled event
+						event := new(scheduler_event.TaskDisabledEvent)
+						event.TaskID = t.id
+						event.Why = fmt.Sprintf("Task disabled with error: %s", t.lastFailureMessage)
+						defer t.eventEmitter.Emit(event)
+						return
+					}
+				} else if consecutiveFailures > 0 {
```

Comment:
>  For example, a publisher may be not available for a long period of time. It will be nice if the servers being monitored can give some breath room for the remote publisher.

:+1: FWIW, that type of failure "buffering" is something we're working on.  An RFC is forthcoming.  I'll make a note to tag you when it lands.